atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/AtlasLockDescriptorUtils.java

```diff
@@ -40,7 +40,7 @@ private AtlasLockDescriptorUtils() {
         TableReference tableRef = tableRefAndRemainder.get().tableRef();
         ByteString remainingBytes = tableRefAndRemainder.get().remainder();
 
-        return IntStream.range(1, remainingBytes.size() - 2)
+        return IntStream.range(1, remainingBytes.size() - 1)
                 .filter(index -> isZeroDelimiterIndex(remainingBytes, index))
                 .mapToObj(index -> createCellFromByteString(remainingBytes, index))
                 .map(cell -> CellReference.of(tableRef, cell))
```

Comment:
I realise this was always an issue (and maybe it's fine we don't fix it initially), but I'm kind of worried about what happens if people write blob keys/columns that have a lot of zero bytes in them. The blow-up factor is bounded by 3001 I think. 

Probably not for this PR, but it may be worth having some warning logs and/or requiring users to turn off a safety flag here.

Reply:
That's a good call. For now, I've added a TODO, but I think we should make sure users are aware.